homeassistant/components/asus_router/__init__.py

```diff
@@ -0,0 +1,71 @@
+"""Native support for Asus wireless routers using HTTP(S) API."""
+
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, PLATFORMS, ROUTER, STOP_LISTENER
+from .router import ARDevice
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+) -> bool:
+    """Set up Asus Router platform."""
+
+    router = ARDevice(hass, config_entry)
+    await router.setup()
+
+    router.async_on_close(config_entry.add_update_listener(update_listener))
+
+    async def async_close_connection(event):
+        """Close router connection of HA stop."""
+
+        await router.close()
+
+    stop_listener = hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP, async_close_connection
+    )
+
+    hass.data.setdefault(DOMAIN, {})[config_entry.entry_id] = {
+        ROUTER: router,
+        STOP_LISTENER: stop_listener,
+    }
+
+    hass.config_entries.async_setup_platforms(config_entry, PLATFORMS)
```

Comment:
Hi There,

This needs to be changed to:

await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)

Reply:
Hey, thanks! I did not know about this before and only found out when getting warnings with the 2023.2 update yesterday